ParkingZoneApp.Tests/Controller/ParkingZoneControllerTests.cs

```diff
@@ -0,0 +1,315 @@
+            Assert.False(_parkingZoneController.ModelState.IsValid);
+        }
+        #endregion
+
+        #region Delete
+        [Fact]
+        public void GivenId_WhenDeleteIsCalled_ThenItShouldReturnNotFoundResult()
+        {
+            //Arrange
+            _service.Setup(service => service.GetById(_parkingZones[0].Id));
+
+            //Act
+            var delete = _parkingZoneController.Delete(_parkingZones[0].Id);
+
+            //Assert
+            Assert.IsType<NotFoundResult>(delete);
+            _service.Verify(service => service.GetById(_parkingZones[0].Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenId_WhenDeleteIsCalled_ThenItShouldReturnDeleteViewModel()
+        {
+            //Arrange
+            _service
+                .Setup(service => service.GetById(_parkingZones[1].Id))
+                .Returns(_parkingZones[1]);
+
+            //Act
+            var delete = _parkingZoneController.Delete(_parkingZones[1].Id);
+            var model = (delete as ViewResult).Model;
+
+            //Assert
+            Assert.IsType<ViewResult>(delete);
+            Assert.IsType<ParkingZone>(model);
+            Assert.Equal(JsonSerializer.Serialize(_parkingZones[1]), JsonSerializer.Serialize(model));
+            Assert.NotNull(delete);
+            Assert.NotNull(model);
+            _service.Verify(service => service.GetById(_parkingZones[1].Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenId_WhenDeleteConfirmedIsCalled_ThenItShouldReturnRedirectToAction()
+        {
+            //Arrange
+            _service
+                .Setup(service => service.GetById(_parkingZones[0].Id))
+                .Returns(_parkingZones[0]);
+
+            _service.Setup(service => service.Delete(_parkingZones[0]));
+
+            //Act
+            var deleteConfirmed = _parkingZoneController.DeleteConfirmed(_parkingZones[0].Id);
+            var model = deleteConfirmed as RedirectToActionResult;
+
+            //Assert
+            Assert.IsType<RedirectToActionResult>(deleteConfirmed);
+            Assert.Equal("Index", model.ActionName);
+            Assert.NotNull(deleteConfirmed);
+            Assert.NotNull(model);
+            _service.Verify(service => service.GetById(_parkingZones[0].Id), Times.Once);
+            _service.Verify(service => service.Delete(_parkingZones[0]), Times.Once);
+        }
+
+        [Fact]
+        public void GivenId_WhenDeleteConfirmedIsCalled_ThenItShouldReturnNotFoundResult()
+        {
+            //Arrange
+            _service.Setup(service => service.GetById(_parkingZones[0].Id));
+
+            //Act
+            var deleteConfirmed = _parkingZoneController.DeleteConfirmed(_parkingZones[0].Id);
+
+            //Assert
+            Assert.IsType<NotFoundResult>(deleteConfirmed);
+            Assert.NotNull(deleteConfirmed);
+            _service.Verify(service => service.GetById(_parkingZones[0].Id), Times.Once);
+        }
+        #endregion
```

Comment:
geneally your tests are good, you did with different point of view, and you have some code not included in others tests